discovery/discovery.go

```diff
@@ -82,28 +78,24 @@ func (o *orchestratorPool) GetOrchestrators(numOrchestrators int) ([]*net.Orches
 		go getOrchInfo(uri)
 	}
 
-	select {
-	case <-ctx.Done():
-		respLock.Lock()
-		if len(orchInfos) < numOrchestrators {
-			numOrchestrators = len(orchInfos)
+	timeout := false
+	infos := []*net.OrchestratorInfo{}
+	nbResp := 0
+	for i := 0; i < len(uris) && len(infos) < numOrchestrators; i++ {
+		select {
+		case info := <-infoCh:
+			infos = append(infos, info)
+			nbResp++
+		case <-errCh:
+			nbResp++
+		case <-ctx.Done():
+			timeout = true
 		}
-		returnOrchs := orchInfos[:numOrchestrators]
-		respLock.Unlock()
-		glog.Info("Done fetching orch info for orchestrators, context timeout: ", len(returnOrchs))
-		cancel()
-		return returnOrchs, nil
-	case <-orchChan:
-		respLock.Lock()
-		if len(orchInfos) < numOrchestrators {
-			numOrchestrators = len(orchInfos)
-		}
-		returnOrchs := orchInfos[:numOrchestrators]
-		respLock.Unlock()
-		glog.Info("Done fetching orch info for orchestrators, numResponses fetched: ", len(returnOrchs))
-		cancel()
-		return returnOrchs, nil
 	}
+	cancel()
+	glog.Infof("Done fetching orch info nb=%d responses=%d/%d timeout=%t",
```

Comment:
`numOrchs` might be a clearer key name than `nb` in the message.

Reply:
Fixed in https://github.com/livepeer/go-livepeer/pull/1403/commits/dc7828a8a4befdd2a90a6b66498687f404901dec